python/orca/tutorial/pytorch/NCF/train_data_loader.py

```diff
@@ -0,0 +1,237 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import time
+import argparse
+import numpy as np
+import pandas as pd 
+import scipy.sparse as sp
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.utils.data as data
+import torch.nn.functional as F 
+from model import NCF
+
+#Step 0: Parameters And Configuration
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dataset", 
+    type=str, 
+    default="ml-1m", 
+    help="dataset name")
+parser.add_argument("--model", 
+    type=str, 
+    default="NeuMF-end", 
+    help="model name")
+parser.add_argument("--main_path", 
+    type=str, 
+    default="./NCF-Data/", 
+    help="main path")
+parser.add_argument("--model_path", 
+    type=str, 
+    default="./models/", 
+    help="model path")
+parser.add_argument("--out", 
+    type=bool, 
+    default=True, 
+    help="save model or not")
+parser.add_argument("--cluster_mode", 
+    type=str, 
+    default="local", 
+    help="")
+parser.add_argument("--lr", 
+    type=float, 
+    default=0.001, 
+    help="learning rate")
+parser.add_argument("--dropout", 
+    type=float, 
+    default=0.0, 
+    help="dropout rate")
+parser.add_argument("--batch_size", 
+    type=int, 
+    default=256, 
+    help="batch size for training")
+parser.add_argument("--epochs", 
+    type=int, 
+    default=20, 
+    help="training epoches")
+parser.add_argument("--top_k", 
+    type=int, 
+    default=10, 
+    help="compute metrics@top_k")
+parser.add_argument("--factor_num", 
+    type=int, 
+    default=32, 
+    help="predictive factors numbers in the model")
+parser.add_argument("--num_layers", 
+    type=int, 
+    default=3, 
+    help="number of layers in MLP model")
+parser.add_argument("--num_ng", 
+    type=int, 
+    default=4, 
+    help="sample negative items for training")
+parser.add_argument("--test_num_ng", 
+    type=int, 
+    default=0, 
+    help="sample part of negative items for testing")

```

Comment:
For many of the arguments, we don't need to modify this, just give a default value in the code.